### DISCUSSION TITLE
esp32-8048S070C can't use pins 19 and 20 since the touch driver uses them exclusively

### DISCUSSION THREAD

**Discussion by DougK71** [rzeldent/esp32-smartdisplay#105] [General]
It with touch I2C bombs at run time even if  pins other than 19 or 20 .....Ex  18 17
Code will compile with the Wire.xxx commands but crashes when it runs 
I know the uart will work and will try SPI using 3 different pins than the LCD SPI and the touch I2C pins

**Comment by rzeldent** [comment]
HI Doug,

You might consider using the low level functions. I think this will work! You can use the i2c port defined in GT911_I2C_HOST.

https://demo-dijiudu.readthedocs.io/en/latest/api-reference/peripherals/i2c.html

**Comment by DougK71** [comment]
It turns out the rs232 could work . The data is small and arrives at a low rate. The project still needs 5v power to the LCD board.
The usb is used for flashing the esp32 in platformio but when not programming the usb connector differential pins could be used for TTLlevel RS232 . The usb cable connection to a PC already emulates a Comm port and provides the 5v and Gnd.
I'll test this out in the next day or two.
As to I2C it could work and avoid the crash if I2C is used via Wire.h but there is the small risk of accidentally writing to the touch GT911 registers and breaking touch. SPI is also an option but it is very busy handling the LCD again I haven't tested it.
Serial has the advantage it is easy to monitor

**Reply by rzeldent** [reply]
And you have another i2C port. GT911 only uses one port....

**Reply by DougK71** [reply]
Well 
I have using the arduino IDE two test arduino boards one esp32 as the receiver and the other an esp8266 as the transmitter
The transmitter does a Serial.println("message"); and the receiver uses serialEvent1() to  get the message. It works.
Now the goal is to get the message into the ESP32-8048S070 touch using uart1
rx will be pin 18 and I have traced from the P4 4 pin connector to the pin on the esp chip.
On my oscilloscope with ground and  pin 18 I see the signal and in platformio code I have Serial1.begin(115200) and code on SerialEvent1() to read in a byte from the buffer exactly like in the test that works.
 This doesn't work with the LCD board ...like as if the pin 18 has not been assigned to Uart1 duty. 
In arduino the serial1.begin assigns the UART1 pins for Uart1 duty but maybe is broken on the ESP32-8048S070 
my message is 8 bytes is sent every 1/5th of a second. Uart0 Serial is used for printing to the platformio monitor so that is why Uart1 is chosen
As to I2C. 
For I2C I understand that any set of pins can be used but if I initialize Wire setting address SDA and SCL pins it compiles but crashes the LCD when uploaded so I don't know how to get this to work

**Comment by DougK71** [comment]
I also tried with another ESP32-8048S070 board just in case  pin 18  was blown

**Reply by rzeldent** [reply]
Have you checked the baud rate?
Also, try to do this with a TTL USB to serial without any display but logging to serial0
Have you called the Serial2.begin(115200, SERIAL_8N1, RXD2, TXD2)?

**Comment by DougK71** [comment]
Yes the baudrate is accurate at 115200 the scope confirmed this 3.3v peak to peak but since in TTL it is inverted the scope did not decode it. The TTL at 115200 is at pin 18 according to the scope The test arduino shows the full correct message in the monitor window on Uart0 that was received on Uart1
The pins have modes and I wonder if the mode isn't being set or RTS DTR is set. 
Nothing is in the received data buffer and a valid signal is on pin18

**Comment by DougK71** [comment]
Further investigation if I use the esp32s3 dev module as the board in Arduino 1.8.18 with code that works with
esp32 D1 mini ....Serial1 doesn't work just as it didn't work with the platformio and the esp32 8048S070.
 So the issue stays with the board or the definition of the pins assigned by invoking Serial1.begin()
board esp32 D1 mini gets the Serial1 correct and esp32s3 dev module is not working as is the platformio.ini
with board = esp32-8048S070C and lib_deps=rzeldent/esp32_smartdisplay
I verified that the ttl signal at 115200 baud makes it to pin18
Any ideas as to fixing this issue.?

**Comment by rzeldent** [comment]
Hi Doug,

Suggest to first test without any display code, just plain arduino to eliminate the possible pin assignments etc...

Additionally, if for the resistive model the IO18 is connected to with a 10k resisted to ground (for the XPT2046). Maybe this is something?
<img width="646" alt="image" src="https://github.com/rzeldent/esp32-smartdisplay/assets/7620231/f37b1480-6caa-4158-9487-8ca8b6c58c92">


maybe try this code, from https://forum.arduino.cc/t/how-to-establish-a-serial-communication-between-an-esp32-s3-dev-kit-c-module-and-arduino-board/1139692/7:

```
// ESP32  Serial1 test - for loopback test connect pins 16 and 17


#define RXD1 16
#define TXD1 17

void setup() {
  // initialize both serial ports:
  Serial.begin(115200);
  Serial1.begin(115200, SERIAL_8N1, RXD1, TXD1);
  Serial.println();
  Serial.println("\n\nESP32 serial1  test Rx pin 16 Tx pin 17");
  Serial.write("   for loopback test connect pin 16 to pin 17\n");
}

void loop() {
  // read from port 1, send to port 0:
  if (Serial1.available()) {
    int inByte = Serial1.read();
    Serial.write(inByte);
  }

  // read from port 0, send to port 1:
  if (Serial.available()) {
    int inByte = Serial.read();
    //Serial.write(inByte);
    Serial1.write(inByte);
  }
}
```

**Reply by DougK71** [reply]
I did try flashing the working code on the recv code below but it did not work with the board
It seems like the board has level shift on pin18 15k to ground 10k to 3.3v like from 5v to 3v  pin 17 is not connected to ground or 3.3v
I have a gt911 touch controller the schematic for the XPT2046 has a 10k pull up for pin 11
the fact that pin18 has a pull down and a pull up might mean it is already used for something else or they are expecting 5v TTL being received
I'll next try an old 5v arduino board and see if all they are doing is level shifting
Thanks for the guidance I now have something to check out


'// dk small profile esp32  WeMos DI mini Esp32 silk screen has Tx and Rx reversed
//          serial 1  rx is pin gpio 09 aka SD2 
//                    tx is pin gpio 10 aka SD3
////    scope expects  RS232 in decode  but TTL inverted 3.3v so decode fails

// for testing we used the esp32 Wemos D1 mini since the 7 inch yellow pcb isn't receiving
//      
//     test with just the CPU esp32 S3 vroom 1  as esp32s3 dev module it fails if
//     this code is uploaded to the 7 inch yelow pcb board 

void setup() {
  // put your setup code here, to run once:
Serial.begin(115200);
Serial1.begin(115200);

// initialize digital pin LED_BUILTIN as an output.
  pinMode(LED_BUILTIN, OUTPUT);
  digitalWrite(LED_BUILTIN, HIGH);    // turn the LED on
  delay(1000);
}

void loop() {
  // put your main code here, to run repeatedly:
 digitalWrite(LED_BUILTIN, LOW);    // turn the LED off by making the voltage LOW
 while ( Serial1.available())
 {
  digitalWrite(LED_BUILTIN, HIGH); 
  char c = Serial1.read();
    Serial.print(c);
 }

}'

**Comment by DougK71** [comment]
looks like pins 18 and 17 are used for the speaker and IIS but there are zero ohm resistor jumpers I removed R5 10 k and the R27 zero ohm so pin 18 is direct to the jumper no pullup or pulldown and it didn't work as Uart1 Rx so the issue likely is deeper in the code and the pins aren't being assigned to Uart mode.
Quite a journey all I needed was to get  a few bytes per second in and to get a message out to reflect screen buttons being pressed. Tomorrow  I try a testing again with a simple program

**Comment by DougK71** [comment]
the 15k pulldown on pin18 is the resistance within the sound chip to ground not a true resistor the pull up R5 was a resistor.
With the removal the screen works fine but rs232 rx still has the issue and there is no path to ground or 3,3v on pin 18
Sunton's schematic could be better but at least there is one.

**Comment by DougK71** [comment]
Further investigation Serial1 doesn't work for output pin17 either . 
Has anyone been able to get data in/out of this board?...I have yet to try SPI..I am wary of using  I2c since the board doesn't label any pins for it and it is used for the touch controller ....but it does has labelled pins for Uart1 but so far no luck in making them work. The S3 chip has special usb features maybe the second usb channel is owning pins 18 and 17 aka Uart1

**Comment by rzeldent** [comment]
WIFI direct?
